src/module/Mailman/src/Mailman/MailmanWorker.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * This file is part of Athene2.
+ *
+ * Copyright (c) 2013-2019 Serlo Education e.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @copyright Copyright (c) 2013-2019 Serlo Education e.V.
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/serlo-org/athene2 for the canonical source repository
+ */
+
+namespace Mailman;
+
+use ClassResolver\ClassResolverAwareTrait;
+use Common\Traits\ObjectManagerAwareTrait;
+use Notification\Entity\NotificationInterface;
+use Zend\EventManager\EventManagerAwareTrait;
+
+class MailmanWorker
+{
+    use ClassResolverAwareTrait;
+    use EventManagerAwareTrait;
+    use ObjectManagerAwareTrait;
+
+    public function run()
+    {
+        $timeout = 60 * 20;
+        ini_set('mysql.connect_timeout', $timeout);
+        ini_set('default_socket_timeout', $timeout);
+
+        $workload = [];
+
+        foreach ($this->getWorkload() as $notification) {
+            $subscriber = $notification->getUser();
+            $id = $subscriber->getId();
+            $workload[$id]['subscriber'] = $subscriber;
+            $workload[$id]['notifications'][] = $notification;
+        }
+
+        foreach ($workload as $data) {
+            $this->getEventManager()->trigger(
+                'notify',
+                $this,
+                [
+                    'notifications' => $data['notifications'],
+                    'user' => $data['subscriber'],
+                ]
+            );
+            // TODO: mark notifications as mailed, maybe see offset from NotificationWorker
```

Comment:
What do do here? We have multiple options:
- Easiest: Send all unseen notifications even if they were sent previously (this would maybe make sense for a weekly batch but probably way too annoying)
- Easy: Somehow persist the last notification that got handled by Mailman. Only consider "newer" notifications (I think this is how the NotificationWorker does things)
- Not as easy (and maybe related to the previous question): add additional field to notifications that marks notifications as mailed

Reply:
see above